TestGeometricSeqPhenotype.java

```diff
@@ -0,0 +1,199 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Map;
+import java.util.Random;
+import java.util.Scanner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * A class for testing the methods in GeometricSeqPhenotype.java
+ *
+ * @author Thien Tran
+ */
+public class TestGeometricSeqPhenotype {
+    // Private fields
+    private GeometricSeqPhenotype emptyPheno; // Default constructor
+    private GeometricSeqPhenotype simplePheno; // Give a specific sequence
+    private GeometricSeqPhenotype simplePheno2; // Give a specific sequence
+
+    GeometricSeqPhenotype [] history;
+
+    /**
+     * Define variables needed for multiple tests.
+     */
+    @Before
+    public void setUp() {
+        // initialize static parameters
+        Parameters.load();
+        Parameters.initialize();
+
+        Random r = new Random();
+        long seed = 5;
+        r.setSeed(seed);
+        emptyPheno = new GeometricSeqPhenotype();
+        simplePheno = new GeometricSeqPhenotype(0.0, 0.0, new char[]{'T', 'G', 'C', 'A', 'T', 'C'});
+
+        simplePheno2 = new GeometricSeqPhenotype(20.0, 10.0, new char[]{'T', 'G', 'C', 'A', 'T', 'C'});
+
+        // Create a small history of GeometricSeq phenotypes
+        history = new GeometricSeqPhenotype[3];
+
+        history[0] = new GeometricSeqPhenotype(1.0, 2.0, new char[]{'T', 'G', 'C', 'A', 'T', 'C'});
+        history[1] = new GeometricSeqPhenotype(3.0, 2.0, new char[]{'T', 'G', 'C', 'A', 'T', 'C'});
+        history[2] = new GeometricSeqPhenotype(5.0, 8.0, new char[]{'T', 'G', 'C', 'A', 'T', 'C'});
+    }
+
+    /**
+     * Test constructors for the SequencePhenotype class.
+     */
+    @Test
+    public void testConstructors() {
+        assertEquals(0, emptyPheno.getSequence().length() % 3);
+        assertEquals(Parameters.startingSequence.length(), emptyPheno.getSequence().length());
+
+        assertEquals("TGCATC", simplePheno.getSequence());
+    }
+
+    /**
+     * Test getter method.
+     */
+    @Test
+    public void testGetTraits() {
+        assertEquals(0.0, simplePheno.getTraitA(), 0);
+        assertEquals(0.0, simplePheno.getTraitB(), 0);
+    }
+
+    /**
+     * Test getter method.
+     */
+    @Test
+    public void testGetSequence() {
+        assertEquals("TGCATC", simplePheno.getSequence());
+    }
+
+    @Test
+    public void testDistance() {
+        assertEquals(2.236068, simplePheno.distance(history[0]), 0.000001);
```

Comment:
Could you please add a few comments describing what you're testing in more detail?